File: app/src/main/java/com/example/alexander/trams/IntegerParser.java
package com.example.alexander.trams;

import org.jsoup.nodes.Element;
import java.util.regex.*;

public class IntegerParser {
    public static Integer getIntFromString(Element e, String word) {
        Pattern pattern = Pattern.compile("\\d+");
        Matcher matcher = pattern.matcher(word);
        int start = 0;
        Integer result = 0;
        while (matcher.find(start)) {
            String value = word.substring(matcher.start(), matcher.end());
            result = Integer.parseInt(value);
            start = matcher.end();
        }
        return result;
    }
}

File: app/src/main/java/com/example/alexander/trams/IntegerParser.kt
package com.example.alexander.trams

import java.util.regex.*

object IntegerParser {
    fun getIntFromString(word: String): Int {
        val pattern = Pattern.compile("\\d+")
        val matcher = pattern.matcher(word)
        var start = 0
        var result = 0
        while (matcher.find(start)) {
            val value = word.substring(matcher.start(), matcher.end())
            result = Integer.parseInt(value)
            start = matcher.end()
        }
        return result
    }
}
